File: openapi/code_samples/java/api_locations_{county}_getAllLocations.java
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/location")
public class LocationController {

    @GetMapping("/getAllCitiesByCounty/{county}")
    public ResponseEntity<PageCityDetailsModel> getAllCitiesByCounty(
            @PathVariable String county,
            @RequestParam(defaultValue = "0") int pageNumber,
            @RequestParam(defaultValue = "50") int pageSize) {
        // Your implementation here to retrieve the list of cities for the given county
        // Simulated data for demonstration purposes
        List<CityDetailsModel> cities = Arrays.asList(
                new CityDetailsModel("City 1", county),
                new CityDetailsModel("City 2", county),
                new CityDetailsModel("City 3", county)
        );

        // Filter cities by the specified county
        List<CityDetailsModel> citiesInCounty = cities.stream()
                .filter(city -> city.getCountyName().equals(county))
                .collect(Collectors.toList());

        // Paginate the results
        List<CityDetailsModel> paginatedCities = citiesInCounty
                .subList(pageNumber * pageSize, Math.min((pageNumber + 1) * pageSize, citiesInCounty.size()));

        PageCityDetailsModel pageCityDetails = new PageCityDetailsModel();
        pageCityDetails.setItems(paginatedCities);
        pageCityDetails.setPageNumber(pageNumber);
        pageCityDetails.setPageSize(pageSize);
        pageCityDetails.setTotalItems(citiesInCounty.size());

        return ResponseEntity.ok(pageCityDetails);
    }
}

@Data
@AllArgsConstructor
class CityDetailsModel {
    private String cityName;
    private String countyName;
}

@Data
class PageCityDetailsModel {
    private List<CityDetailsModel> items;
    private int pageNumber;
    private int pageSize;
    private int totalItems;
}

File: openapi/code_samples/csharp/api_locations_{county}_getAllLocations.cs
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;

[ApiController]
[Route("api/[controller]")]
public class LocationController : ControllerBase
{
    [HttpGet]
    [Route("getAllCitiesByCounty/{county}")]
    [SwaggerOperation(Summary = "Retrieves the list of all locations inside a city.")]
    [ProducesResponseType(typeof(PageCityDetailsModel), 200)]
    [ProducesResponseType(typeof(Fault), 400)]
    public IActionResult GetAllCitiesByCounty(string county, int pageNumber = 0, int pageSize = 50)
    {
        // Your implementation here to retrieve the list of cities for the given county
        // Simulated data for demonstration purposes
        var cities = new List<CityDetailsModel>
        {
            new CityDetailsModel { CityName = "City 1" },
            new CityDetailsModel { CityName = "City 2" },
            new CityDetailsModel { CityName = "City 3" }
            // Add more cities as needed
        };

        // Filter cities by the specified county
        var citiesInCounty = cities.Where(city => city.CountyName == county).ToList();

        // Paginate the results
        var paginatedCities = citiesInCounty.Skip(pageNumber * pageSize).Take(pageSize).ToList();

        PageCityDetailsModel pageCityDetails = new PageCityDetailsModel
        {
            Items = paginatedCities,
            PageNumber = pageNumber,
            PageSize = pageSize,
            TotalItems = citiesInCounty.Count
        };

        return Ok(pageCityDetails);
    }
}

public class CityDetailsModel
{
    public string CityName { get; set; }
    public string CountyName { get; set; }
}

public class PageCityDetailsModel
{
    public List<CityDetailsModel> Items { get; set; }
    public int PageNumber { get; set; }
    public int PageSize { get; set; }
    public int TotalItems { get; set; }
}
